multinet/auth/util.py

```diff
@@ -0,0 +1,44 @@
+"""Utility functions for auth."""
+
+from typing import Any, Optional
+
+from multinet.errors import Unauthorized
+from multinet.types import Workspace
+from multinet.auth.types import UserInfo
+from multinet.user import current_user
+
+
+# NOTE: unfortunately, it is difficult to write a type signature for this
+# decorator. I've opened an issue to ask about this here:
+# https://github.com/python/mypy/issues/9032.
+def require_login(f: Any) -> Any:
+    """Decorate an API endpoint to check for a logged in user."""
+
+    def wrapper(workspace: str, *args: Any, **kwargs: Any) -> Any:
+        user = current_user()
+        if user is None:
+            raise Unauthorized("You must be logged in to perform this action")
+
+        return f(workspace, *args, **kwargs)
+
+    return wrapper
```

Comment:
Fixing my other suggestion, as it was out of date. Sorry it's so large, I wanted to include the import:

```suggestion
from typing import Any, Optional, Callable

from multinet.errors import Unauthorized
from multinet.types import Workspace
from multinet.auth.types import UserInfo
from multinet.user import current_user


# NOTE: unfortunately, it is difficult to write a type signature for this
# decorator. I've opened an issue to ask about this here:
# https://github.com/python/mypy/issues/9032.
def require_login(f: Callable) -> Callable:
    """Decorate an API endpoint to check for a logged in user."""

    def wrapper(*args: Any, **kwargs: Any) -> Any:
        user = current_user()
        if user is None:
            raise Unauthorized("You must be logged in to perform this action")

        return f(*args, **kwargs)

    return wrapper
```